framework/templates.py

```diff
@@ -13,4 +13,5 @@ def populate_properties(user_properties):
 
 
 def fill_template(template, properties):
+    properties["http_content_encoding"] = "$http_content_encoding"
```

Comment:
You should not set any properties here. You must do this for such a client/server. For example:
```
            "id": "nginx",
            "type": "nginx",
            "port": "8000",
            "status_uri": "http://${server_ip}:8000/nginx_status",
            "http_content_encoding": "$http_content_encoding",
```